### PR TITLE
Feature/df 506

### PR DESCRIPTION
**Pull request by ganqqwerty**

### CODE REVIEW

File: alv-portal-ui/pipeline-scripts/translations_needed.csv [hunk 1]
```diff
@@ -0,0 +1,76 @@
+page,key,de,en,fr,it
```
Comment (pavax) [inline alv-portal-ui/pipeline-scripts/translations_needed.csv]: ehmmmmm revert and if needed add it to .gitignore?

File: alv-portal-ui/src/redirection.guard.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  providedIn: 'root'
+})
+export class RedirectionGuard implements CanActivate {
+  constructor(private router: Router) {
+
+  }
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
```
Comment (pavax) [inline alv-portal-ui/src/redirection.guard.ts]: remove the return types that are not needed

File: alv-portal-ui/src/redirection.guard.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    const fragment = state.root.fragment;
+    if (!fragment) {
+      return true;
```
Comment (pavax) [inline alv-portal-ui/src/redirection.guard.ts]: .... return false and navigate to 'home', or not?

File: alv-portal-ui/src/redirection.guard.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    }
+    const regExpMatchArray = fragment.match(/\/job-publication-detail\/(.*)/);
+    if (regExpMatchArray) {
+      return this.router.navigateByUrl('manage-job-ads/' + decodeURIComponent(regExpMatchArray[1]));
+    }
+    return true;
```
Comment (pavax) [inline alv-portal-ui/src/redirection.guard.ts]: .... return false and navigate to 'home', or not?

File: alv-portal-ui/src/redirection.guard.ts (hunk 3 above)
Comment (pavax) [reply alv-portal-ui/src/redirection.guard.ts]: otherwhise we'd navigate to the dummy-component, not?

File: alv-portal-ui/src/redirection.guard.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
+    const fragment = state.root.fragment;
+    if (!fragment) {
+      return true;
+    }
+    const regExpMatchArray = fragment.match(/\/job-publication-detail\/(.*)/);
```
Comment (pavax) [inline alv-portal-ui/src/redirection.guard.ts]: [suggestion] let's make this more re-usable and define an array of legayRedirections
```ts
const legacyUrlRedirections = [
  {
     pattern: '/\/job-publication-detail\/(.*)/'
     navigationUrl: (fragment) => { return // to new url) }
  }
]
```

Comment (ganqqwerty) [reply alv-portal-ui/src/redirection.guard.ts]: why now?

Comment (pavax) [reply alv-portal-ui/src/redirection.guard.ts]: another entry will be needed: 
old -> #/job-fingerprint-redirect?externalId=xxxx
new ->  /job-search/job-fingerprint-redirect?externalId=XXXXXXX

File: alv-portal-ui/src/legacy-url-strategy-redirection-guard.service.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Promise<boolean> | boolean {
+    const fragment = state.root.fragment;
+    if (!fragment) {
+      return this.goHome();
+    }
+    const regExpMatchArray = fragment.match(/\/job-publication-detail\/(.*)/);
+    if (regExpMatchArray) {
+      return this.router.navigateByUrl('manage-job-ads/' + decodeURIComponent(regExpMatchArray[1]));
```
Comment (pavax) [inline alv-portal-ui/src/legacy-url-strategy-redirection-guard.service.ts]: please avoid chaining the result of another navigation to the guard - we had problem before with that use:
```ts
this.router.navigateByUrl('manage-job-ads/' + decodeURIComponent(regExpMatchArray[1]));
return false;
````

File: alv-portal-ui/src/legacy-url-strategy-redirection-guard.service.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+      const regExpMatchArray = fragment.match(redirectionRule.pattern);
+      if (regExpMatchArray) {
+        return redirectionRule.action.call(this, next, state, fragment);
+      }
+    }
+    return this.goHome();
+  }
+
+  jobPublicationDetailRedirect(next, state, fragment) {
+    const regExpMatchArray = fragment.match(/\/job-publication-detail\/(.*)/);
```
Comment (99999989) [inline alv-portal-ui/src/legacy-url-strategy-redirection-guard.service.ts]: Isn't that a repetition of line 32?

**Pull request merged**